src/Symfony/Component/Routing/Matcher/UrlMatcher.php

```diff
@@ -186,14 +186,10 @@ protected function matchCollection($pathinfo, RouteCollection $routes)
             }
 
             $hasRequiredScheme = !$route->getSchemes() || $route->hasScheme($this->context->getScheme());
-            if ($requiredMethods) {
-                if (!\in_array($method, $requiredMethods)) {
-                    if ($hasRequiredScheme) {
-                        $this->allow = array_merge($this->allow, $requiredMethods);
-                    }
+            if ($hasRequiredScheme && $requiredMethods && !\in_array($method, $requiredMethods)) {
+                $this->allow = array_merge($this->allow, $requiredMethods);
 
-                    continue;
```

Comment:
this `continue` outside the `if ($hasRequiredScheme)` was preventing to collect the `$this->allowSchemes` bellow